neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/transaction/impl/Neo4jTransactionCoordinatorBuilder.java

```diff
@@ -30,11 +31,14 @@ public Neo4jTransactionCoordinatorBuilder(TransactionCoordinatorBuilder delegate
 
 	@Override
 	public TransactionCoordinator buildTransactionCoordinator(TransactionCoordinatorOwner owner, TransactionCoordinatorOptions options) {
-		TransactionCoordinator coordinator = delegate.buildTransactionCoordinator( owner, options );
 		if ( delegate.isJta() ) {
+			TransactionCoordinator coordinator = delegate.buildTransactionCoordinator( owner, options );
 			return new Neo4jJtaTransactionCoordinator( coordinator, datastoreProvider );
 		}
 		else {
+			// We use a no-op owner to make sure that flushes and events are triggered as expected.
+			NoopJdbcResourceTransactionCoordinatorOwner noopOwner = new NoopJdbcResourceTransactionCoordinatorOwner( owner );
```

Comment:
I'm wondering how the no-op impl plays in here. Shouldn't there be a "real" implementation in case of Neo4j Embedded, just not based on JDBC but on the Neo4j TX API?